src/main/java/seedu/address/logic/commands/EditPropertyCommand.java

```diff
@@ -0,0 +1,348 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_POSTAL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REMARK;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.name.Name;
+import seedu.address.model.property.Address;
+import seedu.address.model.property.Deadline;
+import seedu.address.model.property.PostalCode;
+import seedu.address.model.property.Property;
+import seedu.address.model.property.Type;
+import seedu.address.model.property.client.AskingPrice;
+import seedu.address.model.property.client.Client;
+import seedu.address.model.property.client.Contact;
+import seedu.address.model.property.client.Email;
+import seedu.address.model.remark.Remark;
+
+/**
+ * Edits a property in the app.
+ */
+public class EditPropertyCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit property";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits a property in the app. \n"
+            + "Parameters: INDEX"
+            + PREFIX_NAME + "NAME "
+            + PREFIX_TYPE + "TYPE "
+            + PREFIX_ADDRESS + "ADDRESS "
+            + PREFIX_POSTAL + "POSTAL "
+            + PREFIX_DEADLINE + "DEADLINE "
+            + PREFIX_REMARK + "REMARK "
+            + PREFIX_CLIENT_NAME + "CLIENT_NAME "
+            + PREFIX_CLIENT_CONTACT + "CLIENT_CONTACT "
+            + PREFIX_CLIENT_EMAIL + "CLIENT_EMAIL "
+            + PREFIX_CLIENT_ASKING_PRICE + "CLIENT_ASKING_PRICE \n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Mayfair "
+            + PREFIX_TYPE + "Condo "
+            + PREFIX_ADDRESS + "1 Jurong East Street 32 "
+            + PREFIX_POSTAL + "609477 "
+            + PREFIX_DEADLINE + "31-12-2021 "
+            + PREFIX_REMARK + "Urgent to sell "
+            + PREFIX_CLIENT_NAME + "Alice "
+            + PREFIX_CLIENT_CONTACT + "91234567 "
+            + PREFIX_CLIENT_EMAIL + "alice@gmail.com "
+            + PREFIX_CLIENT_ASKING_PRICE + "$800,000";
+
+    public static final String MESSAGE_SUCCESS = "Edited property: %1$s";
+    public static final String MESSAGE_DUPLICATE_PROPERTY = "This property already exists in the app";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    private final Index index;
+    private final EditPropertyDescriptor editPropertyDescriptor;
+
+    /**
+     * Creates an EditPropertyCommand to edit the specified {@code Property}.
+     */
+    public EditPropertyCommand(Index index, EditPropertyCommand.EditPropertyDescriptor editPropertyDescriptor) {
```

Comment:
All the `EditPropertyCommand.EditPropertyDescriptor` can be simplified to just `EditPropertyDescriptor` since `EditPropertyDescriptor` is an inner class of `EditPropertyCommand`